diff --git a/MAQSS_GCS.pro b/MAQSS_GCS.pro
index 93b1c69..fc2849b 100644
--- a/MAQSS_GCS.pro
+++ b/MAQSS_GCS.pro
@@ -3,8 +3,8 @@ QT += qml gui quick positioning location
 CONFIG += c++11
 
 SOURCES += main.cpp \
-    ../xbeeplus/lib/ReceivePacket.cpp \
-    ../xbeeplus/lib/SerialXbee.cpp \
+    ../xbeeplus/src/ReceivePacket.cpp \
+    ../xbeeplus/src/SerialXbee.cpp \
 #    ../../NGCP/xbeeplus/lib/TransmitRequest.cpp \
 #    ../../NGCP/xbeeplus/lib/Utility.cpp \
     XbeeInterface.cpp
diff --git a/MAQSS_GCS.pro.user b/MAQSS_GCS.pro.user
index 47e28e0..d6a1e84 100644
--- a/MAQSS_GCS.pro.user
+++ b/MAQSS_GCS.pro.user
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE QtCreatorProject>
-<!-- Written by QtCreator 4.2.1, 2017-01-31T22:16:20. -->
+<!-- Written by QtCreator 4.0.2, 2017-03-08T20:11:28. -->
 <qtcreator>
  <data>
   <variable>EnvironmentId</variable>
@@ -54,21 +54,19 @@
  </data>
  <data>
   <variable>ProjectExplorer.Project.PluginSettings</variable>
-  <valuemap type="QVariantMap">
-   <valuelist type="QVariantList" key="ClangStaticAnalyzer.SuppressedDiagnostics"/>
-  </valuemap>
+  <valuemap type="QVariantMap"/>
  </data>
  <data>
   <variable>ProjectExplorer.Project.Target.0</variable>
   <valuemap type="QVariantMap">
-   <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Desktop Qt 5.8.0 GCC 64bit</value>
-   <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName">Desktop Qt 5.8.0 GCC 64bit</value>
-   <value type="QString" key="ProjectExplorer.ProjectConfiguration.Id">qt.58.gcc_64_kit</value>
+   <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Desktop Qt 5.7.0 GCC 64bit</value>
+   <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName">Desktop Qt 5.7.0 GCC 64bit</value>
+   <value type="QString" key="ProjectExplorer.ProjectConfiguration.Id">qt.57.gcc_64_kit</value>
    <value type="int" key="ProjectExplorer.Target.ActiveBuildConfiguration">0</value>
    <value type="int" key="ProjectExplorer.Target.ActiveDeployConfiguration">0</value>
    <value type="int" key="ProjectExplorer.Target.ActiveRunConfiguration">0</value>
    <valuemap type="QVariantMap" key="ProjectExplorer.Target.BuildConfiguration.0">
-    <value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">/home/oem/Documents/GCS/build-MAQSS_GCS-Desktop_Qt_5_8_0_GCC_64bit-Debug</value>
+    <value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">/home/oem/Documents/GCS/build-MAQSS_GCS-Desktop_Qt_5_7_0_GCC_64bit-Debug</value>
     <valuemap type="QVariantMap" key="ProjectExplorer.BuildConfiguration.BuildStepList.0">
      <valuemap type="QVariantMap" key="ProjectExplorer.BuildStepList.Step.0">
       <value type="bool" key="ProjectExplorer.BuildStep.Enabled">true</value>
@@ -128,7 +126,7 @@
     <value type="bool" key="Qt4ProjectManager.Qt4BuildConfiguration.UseShadowBuild">true</value>
    </valuemap>
    <valuemap type="QVariantMap" key="ProjectExplorer.Target.BuildConfiguration.1">
-    <value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">/home/oem/Documents/GCS/build-MAQSS_GCS-Desktop_Qt_5_8_0_GCC_64bit-Release</value>
+    <value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">/home/oem/Documents/GCS/build-MAQSS_GCS-Desktop_Qt_5_7_0_GCC_64bit-Release</value>
     <valuemap type="QVariantMap" key="ProjectExplorer.BuildConfiguration.BuildStepList.0">
      <valuemap type="QVariantMap" key="ProjectExplorer.BuildStepList.Step.0">
       <value type="bool" key="ProjectExplorer.BuildStep.Enabled">true</value>
@@ -188,7 +186,7 @@
     <value type="bool" key="Qt4ProjectManager.Qt4BuildConfiguration.UseShadowBuild">true</value>
    </valuemap>
    <valuemap type="QVariantMap" key="ProjectExplorer.Target.BuildConfiguration.2">
-    <value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">/home/oem/Documents/GCS/build-MAQSS_GCS-Desktop_Qt_5_8_0_GCC_64bit-Profile</value>
+    <value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">/home/oem/Documents/GCS/build-MAQSS_GCS-Desktop_Qt_5_7_0_GCC_64bit-Profile</value>
     <valuemap type="QVariantMap" key="ProjectExplorer.BuildConfiguration.BuildStepList.0">
      <valuemap type="QVariantMap" key="ProjectExplorer.BuildStepList.Step.0">
       <value type="bool" key="ProjectExplorer.BuildStep.Enabled">true</value>
@@ -312,7 +310,7 @@
     <value type="QString" key="Qt4ProjectManager.Qt4RunConfiguration.ProFile">MAQSS_GCS.pro</value>
     <value type="bool" key="Qt4ProjectManager.Qt4RunConfiguration.UseDyldImageSuffix">false</value>
     <value type="QString" key="Qt4ProjectManager.Qt4RunConfiguration.UserWorkingDirectory"></value>
-    <value type="QString" key="Qt4ProjectManager.Qt4RunConfiguration.UserWorkingDirectory.default">/home/oem/Documents/GCS/build-MAQSS_GCS-Desktop_Qt_5_8_0_GCC_64bit-Debug</value>
+    <value type="QString" key="Qt4ProjectManager.Qt4RunConfiguration.UserWorkingDirectory.default"></value>
     <value type="uint" key="RunConfiguration.QmlDebugServerPort">3768</value>
     <value type="bool" key="RunConfiguration.UseCppDebugger">false</value>
     <value type="bool" key="RunConfiguration.UseCppDebuggerAuto">true</value>
diff --git a/MainPage.qml b/MainPage.qml
index c8fb6a3..d3a3ab2 100644
--- a/MainPage.qml
+++ b/MainPage.qml
@@ -181,6 +181,7 @@ Item {
 
         // else, create a new Quadcopter object
         if (!done_flag && component.status === Component.Ready) {
+
             quadcopters[nextNdx] = component.createObject(mapContainer, {"idNumber": msg_container.vehicleID, "coordLLA": msg_container.vehicleLocation})
 
             // warn object creation error
diff --git a/XbeeInterface.cpp b/XbeeInterface.cpp
index b92bbbd..ef80ec1 100644
--- a/XbeeInterface.cpp
+++ b/XbeeInterface.cpp
@@ -82,6 +82,7 @@ void XbeeInterface::startComms() {
 
     qDebug() << "Starting Serial Interface";
     serial_interface.Connect();
+    qDebug() << "Connected";
     serial_interface.AsyncReadFrame();
     comms = true;
 }
diff --git a/XbeeInterface.hpp b/XbeeInterface.hpp
index 3204519..aadd7f7 100644
--- a/XbeeInterface.hpp
+++ b/XbeeInterface.hpp
@@ -5,7 +5,7 @@
 #include <QDebug>
 #include <QString>
 #include <QQmlContext>
-//#include "../xbeeplus/include/SerialXbee.hpp"
+#include "../xbeeplus/include/SerialXbee.hpp"
 #include "../xbeeplus/include/ReceivePacket.hpp"
 #include "../xbeeplus/include/TransmitRequest.hpp"
 
diff --git a/js_resources/Coordinates.js b/js_resources/Coordinates.js
index e0c9a57..16aad4a 100644
--- a/js_resources/Coordinates.js
+++ b/js_resources/Coordinates.js
@@ -153,7 +153,7 @@ function writeMsnString(searchChunks, vehicleCoords, choices, cornerChoices, fie
         // Generate message string following format in heading "NEWMSG,MSN,Q1,P35.123456 120.123456 0.0000,H1.23456,F139.7,D100.4567"
         msnString[i1] = "NEWMSG" + ",MSN"
                 + ",Q" + mainPage.quadcopters[choices[i1]].idNumber
-                + ",P" + String(startLocs[i1][0]) + ":" + String(startLocs[i1][1]) + ":" + String(startLocs[i1][2])
+                + ",P" + String(startLocs[i1][0]) + " " + String(startLocs[i1][1]) + " " + String(startLocs[i1][2])
                 + ",H" + String(headings[i1])
                 + ",F" + String(fieldHeading)
                 + ",D" + String(distance);
diff --git a/js_resources/Messaging.js b/js_resources/Messaging.js
index 1c56b90..69736fc 100644
--- a/js_resources/Messaging.js
+++ b/js_resources/Messaging.js
@@ -57,7 +57,7 @@ function handleGenericMsg(msg) {
         // Interpret
         console.log("Handle UPDT Msg");
         msg_container.vehicleID = parseInt(split_msg[2][1]);
-        msg_container.vehicleLocation = Utils.strToFloat((split_msg[3].substr(1)).split(":"));
+        msg_container.vehicleLocation = Utils.strToFloat((split_msg[3].substr(1)).split(" "));
         msg_container.vehicleStatus = split_msg[4].substr(1);
         msg_container.vehicleRole = parseInt(split_msg[5][1]);
         console.log("Reported Role: " , msg_container.vehicleRole);
@@ -88,7 +88,7 @@ function generatePOIMsg(msg_container, targetQuad) {
     var msg;
     // check targetQuad is not msg_container.vehicleID (quick vehicle cannot also be detailed vehicle)
     if (msg_container.vehicleID !== targetQuad)
-     msg = "NEWMSG,POI,Q" + msg_container.targetQuad + ",P" + msg_container.targetLocation[0] + ":" + msg_container.targetLocation[1]
+     msg = "NEWMSG,POI,Q" + msg_container.targetQuad + ",P" + msg_container.targetLocation[0] + " " + msg_container.targetLocation[1]
 
     return msg;
 }
